Whisper audio processor

rohansjoshi opened the pull request:
Summary:
Running Whisper to convert audio -> text consists of two steps: 
(1) Audio preprocessor (aka Mel spectrogram feature extractor), and 
(2) Whisper model (encoder+decoder)
Currently, in examples/qualcomm/oss_scripts/whisper, we have a flow for exporting Whisper encoder+decoder (2) and running it on device. It can take in Mel spectrogram tensors as input and produce text output

**This class implements part (1), the audio processing stage, in PyTorch.** It is equivalent to HuggingFace WhisperFeatureExtractor (which computes Mel spectrograms with NumPy). It takes in an audio waveform at 16KHz (as a 1D tensor) and outputs Mel spectrograms that can be fed directly to the Whisper model (2).

The script (see test plan) compares this class agains WhisperFeatureExtractor, they have a very small numerical discrepancy (<1e-5).

Differential Revision: D80215714

cccclai reviewed (approved):
Please update the output file in the unit test and it's all good

mergennachin commented:
cc @msluszniak @chmjkb @mkopcins FYI the audio preprocessing part can be now just exported

rohansjoshi committed:
Whisper audio processor (#13538)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/13538

Running Whisper to convert audio -> text consists of two steps:
(1) Audio preprocessor (aka Mel spectrogram feature extractor), and
(2) Whisper model (encoder+decoder)
Currently, in examples/qualcomm/oss_scripts/whisper, we have a flow for exporting Whisper encoder+decoder (2) and running it on device. It can take in Mel spectrogram tensors as input and produce text output

**This class implements part (1), the audio processing stage, in PyTorch.** It is equivalent to HuggingFace WhisperFeatureExtractor (which computes Mel spectrograms with NumPy). It takes in an audio waveform at 16KHz (as a 1D tensor) and outputs Mel spectrograms that can be fed directly to the Whisper model (2).

The script (see test plan) compares this class agains WhisperFeatureExtractor, they have a very small numerical discrepancy (<1e-5).

Reviewed By: jackzhxng

Differential Revision: D80215714
